include/ginkgo/core/matrix/dense.hpp

```diff
@@ -1012,64 +1012,64 @@ class Dense
     /**
      * @copydoc scale(const LinOp *)
      *
-     * @note  Other implementations of dense should override this function
-     *        instead of scale(const LinOp *alpha).
+     * @deprecated  This function will be removed in the future,
```

Comment:
Yeah, I was also a bit unsure about this. My argument here is: Even if you derive from dense, every class taking Dense inputs will use `values_` and `stride_` to determine how the data will be accessed, so there is nothing sensible behavior-wise that a user could add here. It might make sense to add a logger event to member functions like these (not individual events but a generic linop_member_called (the name is bad, I know 😆 ) with a string parameter for the actual function name.